### PR TITLE
Clarify attributes on Message config topic

### PR DESCRIPTION
**Pull request by Santoshziffity**
## Purpose of this pull request

This pull request (PR) for Clarify attributes on Message config topic like schema, request, publishers, consumers.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/extension-dev-guide/message-queues/config-mq.html

### CODE REVIEW

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md [hunk 1]
```diff
@@ -57,9 +57,9 @@ The `name` parameter is required. The topic definition must include either a `re
 Parameter | Description
 --- | ---
 name | A string that uniquely identifies the topic. A topic name should be a series of strings that are separated by periods. The leftmost string should be the most general, and each string afterward should narrow the scope. For example, to describe actions for tending to pets, you might create names such as `cat.white.feed` and `dog.retriever.walk`. Wildcards are not supported in the `communication.xml` file.
-request | Specifies the data type of the topic.
+request | Specifies the data structure that’s used to define a message and is needed when publishing any messages through Magento.
```
Comment (BarnyShergold) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:60]: Please avoid contractions
```suggestion
request | Specifies the data structure that is used to define a message and is needed when publishing any messages through Magento.
```

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md [hunk 2]
```diff
@@ -57,9 +57,9 @@ The `name` parameter is required. The topic definition must include either a `re
...
 response | Specifies the format of the response. This parameter is required if you are defining a synchronous topic. Omit this parameter if you are defining an asynchronous topic.
-schema | The interface that describes the structure of the message. The format must be  `<module>\Api\<ServiceName>::<methodName>`.
+schema | Schema describes the message structure. Use the Data Interface format from the Service Contracts. Besides, there is another way to specify a service method signature:`<module>\Api\<ServiceName>::<methodName>` (Magento\Customer\Api\CustomerRepositoryInterface::save).
```
Comment (BarnyShergold) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:62]: Avoid conversational language
```suggestion
schema | Schema describes the message structure. Use the Data Interface format from the Service Contracts. There is an alternative way to specify a service method signature:`<module>\Api\<ServiceName>::<methodName>` (E.g. Magento\Customer\Api\CustomerRepositoryInterface::save).
```

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md [hunk 3]
```diff
@@ -73,6 +73,9 @@ method | The method this handler executes.
 disabled | Determines whether this handler is disabled. The default value is `false`.
 
 ### `queue_consumer.xml` {#queueconsumerxml}
+
+As for consumer elements, they map the message receiver to a specific queue. Such parameters as class and method indicate what elements of the Magento 2 Message Queue Framework receive and process the message.
```
Comment (BarnyShergold) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:77]:
```suggestion
With consumer elements, they map the message receiver to a specific queue. Such parameters as class and method indicate what elements of the Magento 2 Message Queue Framework receive and process the message.
```

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md [hunk 4]
```diff
@@ -228,6 +231,8 @@ The following illustrates an `arguments` block:
 
 ### `queue_publisher.xml` {#queuepublisherxml}
 
+The publisher element performs two vital actions: iit configures the type of connection and the same time sets up the exchange to publish to. Magento 2 uses only one exchange out of the box, but multiple exchanges are possible. As for the name of every exchange, it is a part of publisher configuration.
```
Comment (BarnyShergold) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:234]:
```suggestion
The publisher element performs two vital actions, it configures the type of connection and at the same time, sets up the exchange to publish to. Magento uses only one exchange out of the box, but multiple exchanges are possible. The name for every exchange is contained in the publisher configuration.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
I've corrected some of the language used. Please check that you are happy the meaning is still intact.

Good clarifications

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md (hunk 1 above)
Comment (keharper) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:60]:
```suggestion
request | Specifies the data structure that’s used to define a message. It is needed when publishing any messages through Magento.
```

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md (hunk 3 above)
Comment (keharper) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:77]:
```suggestion
Consumer elements map the message receiver to a specific queue. Parameters such as class and method indicate which elements of the Message Queue Framework receive and process the message.
```

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md [hunk 5]
```diff
@@ -248,7 +253,7 @@ The `queue_publisher.xml` file defines which connection and exchange to use to p
 ```
 
 #### `publisher` element
-{:.no_toc}
+{:.no_toc}
```
Comment (keharper) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:256]: Delete the added spaces at the end of this line

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md (hunk 4 above)
Comment (keharper) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:234]:
```suggestion
The publisher element performs two vital actions: 

*  It configures the type of connection.
*  It sets up the exchange to publish to. 

Magento uses only one exchange out of the box, but multiple exchanges are possible. The name of every exchange is a part of the publisher configuration.
```

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md (hunk 2 above)
Comment (keharper) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:62]:
```suggestion
schema | Describes the message structure. Use the Data Interface format from the Service Contracts. Besides, there is another way to specify a service method signature:`<module>\Api\<ServiceName>::<methodName>` (Magento\Customer\Api\CustomerRepositoryInterface::save).
```

Comment (keharper) [reply src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:62]: I don't understand "Use the Data Interface format from the Service Contracts. Besides, there is another way to specify a service method signature:"  Is `<module>\Api\<ServiceName>::<methodName>` not the preferred way to specify a schema name? You don't list the data interface format.

**Review by keharper — CHANGES_REQUESTED** [review]

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md [hunk 6]
```diff
@@ -57,9 +57,9 @@ The `name` parameter is required. The topic definition must include either a `re
...
 --- | ---
 name | A string that uniquely identifies the topic. A topic name should be a series of strings that are separated by periods. The leftmost string should be the most general, and each string afterward should narrow the scope. For example, to describe actions for tending to pets, you might create names such as `cat.white.feed` and `dog.retriever.walk`. Wildcards are not supported in the `communication.xml` file.
-request | Specifies the data type of the topic.
+request | Specifies the data structure that’s used to define a message. It is needed when publishing any messages through Magento.
 response | Specifies the format of the response. This parameter is required if you are defining a synchronous topic. Omit this parameter if you are defining an asynchronous topic.
-schema | The interface that describes the structure of the message. The format must be  `<module>\Api\<ServiceName>::<methodName>`.
+schema | Describes the message structure. Use the Data Interface format from the Service Contracts. Besides, there is another way to specify a service method signature:`<module>\Api\<ServiceName>::<methodName>` (Magento\Customer\Api\CustomerRepositoryInterface::save).
```
Comment (keharper) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:62]: My comment on this line wasn't addressed. You don't mention what the Data Interface format is or differentiate it from `<module>\Api\<ServiceName>::<methodName>`.

**Comment by keharper** [comment]
Closing this PR because it contains content plagiarized from [Firebear Studio](https://firebearstudio.com/blog/magento-2-message-queue-framework.html).

**Pull request closed without merging**